### DISCUSSION TITLE
A easy way to create, or remove, relations of collections, using graphql

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#11773] [Feature Requests]
Good day,

for instance there are 2 collections, customers, and orders,
customers collection contains a many to many with orders,
currently its not possible to easily create, remove, a order for customers,

at the moment when one wants to create a new order inside customers, one would need to do a query on customers to get the ids of the current set relations, and then would need to do a update mutation on customers with the ids of the already created orders + plus the new order.. example:
```
query getOrderIds{
  customers{
    id
    orders{
     orders_id{
        id
      }
    }
  }
}

```

```
mutation newOrder{
  update_customers_item(id:1, data:{
    orders:[{
      orders_id:{
        test: "new Order"
      }
    }
      {
      orders_id:{
        id: 3
      }
    }
    ]}){
    id
  }
}
```

if one wants to remove a order from customer, one would again need to query all the already set orders ids, filter out the id one does not want and then do a update mutation on customers.

this is not ideal. maybe something like this for adding and removing would be abit more user friendly.. 

```
mutation newOrder{
  update_customers_item(id:1, data:{
    orders:{
     remove:[{
         orders_id:{ id: 2}
       }]
     create:[{
         orders_id:{
         test: "new Order" }
       }
      ]
    }){
    id
  }
}
```

kind regards

**Comment by rijkvanzanten** [comment]
Another wonderful case of GraphQLs screaming limitation when it comes to union input types 😭 I wish we could allow for both at the same time, but you can't do that in GraphQL...

I think the next best thing would be to add an additional suffix like `delta` to allow these alternative update options, like:

```graphql
# Current:
mutation newOrder {
  update_customers_item(
    id: 1
    data: {
      orders: [{ orders_id: { test: "new Order" } }, { orders_id: { id: 3 } }]
    }
  ) {
    id
  }
}

# New addition:
mutation newOrder {
  update_customers_item(
    id: 1
    data: {
      orders_delta: { create: [{ test: "new Order" }], delete: [3] }
    }
  ) {
    id
  }
}
```

**Reply by cpt-n3mo** [reply]
> Another wonderful case of GraphQLs screaming limitation when it comes to union input types

Indeed this is a shame..
Your solution seems nice though!

Im i correct that the JS SDK suffers from the same thing..?
As the following also does not seem to Add, only replace the current relations.?

``` javascript
const newOrder = await this.customers.updateOne(5, {orders: [
					{
						orders_id: {
							orderDate: "someDate",
						},
					},
				],
			});

```
 
kind regards!

**Reply by rijkvanzanten** [reply]
> As the following also does not seem to Add, only replace the current relations.?

The SDK uses the REST API internally, which actually already supports a syntax very similar to what you proposed above. See the section under "detailed" on https://docs.directus.io/reference/introduction/#relational-data. This can be used in the SDK as well 🙂 

This feature request is mostly to find a way to expose that same syntax in GraphQL

**Reply by cpt-n3mo** [reply]
>See the section under "detailed"

Perfect! indeed this seems to work perfectly.!  Thx! 

>This feature request is mostly to find a way to expose that same syntax in GraphQL

It would indeed be nice if something like this could be implemented in graphql, 
not sure how difficult it would be to implement the idea though , i have been going though the source, but typescript and graphql typing is just a bit to match type deceleration for me to follow hehe :P .. 
so im not sure how match i can help `finding a way`  next to doing the feature request sorry..

I am very happy to see though,  that the functionality is there within the sdk and rest api!  so #goodlife

kind regards!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
